File: src/main/java/com/medical/medical_chekup/service/DoctorService.java
package com.medical.medical_chekup.service;

import com.medical.medical_chekup.dto.DoctorListItemDTO;
import com.medical.medical_chekup.dto.response.ApiResponsePagination;

public interface DoctorService {

    ApiResponsePagination<DoctorListItemDTO> getAllDoctor(String location, String doctorName, String keyword,
            String treatment,
            Integer current, Integer size);
}

File: frontend/src/services/doctor.service.ts
import { http } from './http'
import type { ApiResponsePagination } from '../types/api'
import type { DoctorListItemDTO } from '../types/doctor'

interface DoctorFilter {
    location?: string
    doctorName?: string
    keyword?: string
    treatment?: string
    current: number
    size: number
}

export async function getAllDoctor(filter: DoctorFilter): Promise<ApiResponsePagination<DoctorListItemDTO> | null> {
    const response = await http.get<ApiResponsePagination<DoctorListItemDTO>>('/doctor/', { params: filter })
    return response.data ?? null
}